src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient_.cs

```diff
@@ -36,7 +60,139 @@ internal partial class ImageStoreClient
                 serverTimeout,
                 cancellationToken);
         }
+       
+        /// <inheritdoc />
+        public Task UploadFileChunkAsync(
+            byte[] fileChunkToUpload,
+            string pathInImageStore,
+            Guid? sessionId,
+            long startBytePosition,
+            long endBytePosition,
+            long length,
+            long? serverTimeout = 60,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return this.UploadFileChunkAsync(
+                fileChunkToUpload,
+                pathInImageStore,
+                sessionId,
+                endBytePosition,
+                endBytePosition,
+                length,
+                null,
+                serverTimeout,
+                cancellationToken);
+        }
+        
+        /// <inheritdoc />
+        public async Task UploadApplicationPackageAsync(
+            string applicationPackagePath,
+            bool compressPackage = false,
+            string applicationPackagePathInImageStore = default(string),
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            applicationPackagePath.ThrowIfNull(nameof(applicationPackagePath));
+
+            var store = await GetImageStoreConnectionString();
+            if (IsLocalStore(store))
+            {
+                var requestId = Guid.NewGuid();
+                ServiceFabricHttpClientEventSource.Current.InfoMessage($"{this.httpClient.ClientId}:{requestId}",
+                    "Processing call for ApplicationClient.DeleteApplicationAsync");
+
+                var absPkgPath = FileUtilities.GetAbsolutePath(applicationPackagePath);
+
+                var pkgPathInImageStore = applicationPackagePathInImageStore;
+                if (string.IsNullOrEmpty(pkgPathInImageStore))
+                {
+                    pkgPathInImageStore = applicationPackagePath.Replace(Path.GetDirectoryName(applicationPackagePath), "");
+                }
+
+                pkgPathInImageStore = pkgPathInImageStore.Trim('\\', '/');
+
+                if (!Directory.Exists(applicationPackagePath))
+                {
+                    throw new InvalidOperationException($"Application package path {applicationPackagePath} not found.");
+                }
+
+                var files = Directory.EnumerateFiles(absPkgPath, "*", SearchOption.AllDirectories)
+                    .Select(file => new System.IO.FileInfo(file));
```

Comment:
This code is duplicated in the else block below, please move it outside of if- else